File: eTravelAgencija.Model/model/Payment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace eTravelAgencija.Model.model
{
    public partial class Payment
    {
        public int ReservationId { get; set; }
        public Reservation reservation {get; set;}
        public int RateId { get; set; }
        public Rate rate {get; set;}
        public decimal Amount { get; set; }
        public DateTime PaymentDate { get; set; }
        public string PaymentMethod { get; set; }
        public DateTime PaymentDeadline { get; set; }
        public bool IsConfirmed { get; set; }
    }
}

File: UI/etravel_desktop/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int reservationId;
  final int rateId;
  final double amount;
  final DateTime paymentDate;
  final String paymentMethod;
  final DateTime paymentDeadline;
  final bool isConfirmed;
  final bool deadlineExtended;

  Payment({
    required this.reservationId,
    required this.rateId,
    required this.amount,
    required this.paymentDate,
    required this.paymentMethod,
    required this.paymentDeadline,
    required this.isConfirmed,
    required this.deadlineExtended,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
